File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> set = new HashMap<>();
        int i = 0;
        int[] ans = new int[2];
        
        for(int num : nums) {
            int diff = target - num;
            if(set.containsKey(diff)) {
                ans[0] = set.get(diff);
                ans[1] = i;
                return ans;
            } else {
                set.put(num, i);
                i++;
            }
        }
        
        return new int[]{-1,-1};
    }
}

File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        vector<int> answer;
        for(auto firstCount = 0; firstCount < nums.size(); ++firstCount) {
            for(auto secondCount = firstCount + 1; secondCount < nums.size(); ++secondCount) {
                if(target == (nums[firstCount] + nums[secondCount])) {
                    answer.push_back(firstCount);
                    answer.push_back(secondCount);
                    return answer;
                }
            }
        }
        
        
        return answer;
        
    }
};